app/services/decision-report-generation.js

```diff
@@ -16,53 +17,55 @@ export default class DecisionReportGeneration extends Service {
         timeOut: 10 * 60 * 1000,
       }
     );
-    const batchedReports = this.batch(reports, 10);
-    let updatedReports = [];
-    let failedReports = [];
-    for (const batch of batchedReports) {
-      await Promise.all(
-        batch.map(async (report) => {
-          try {
-            await this.generateSinglePdf.perform(
-              report,
-              'generate-decision-report'
-            );
-            await this.reloadFile(report);
-            updatedReports.push(report);
-          } catch (error) {
-            failedReports.push({ report, error });
-          }
+    try {
+      const job = await this.generateMultiplePdfs.perform(
+        reports,
+        'generate-decision-report'
+      );
+      this.pollReplacementReports.perform(job, reports, generatingPDFsToast);
+    } catch (error) {
+      this.toaster.error(
+        this.intl.t('error-while-generating-report-pdfs', {
+          error: error.message,
         })
       );
     }
-    this.toaster.close(generatingPDFToast);
-    if (failedReports.length) {
-      for (const failure of failedReports) {
+  });
+
+  pollReplacementReports = task(async (job, reports, generatingPDFsToast) => {
+      const jobResult = await this.getJob.perform(
+        job,
+        'generate-decision-report'
+      );
+      if (jobResult) {
+        if (jobResult.status === CONSTANTS.DECISION_REPORT_JOB_STATUSSES.SUCCESS) {
+          await this.reloadFiles(reports);
+          this.toaster.close(generatingPDFsToast);
+          this.toaster.success(
+            this.intl.t(
+              'decision-report-generation--toast-generating-complete--message',
+              {
+                total: reports.length,
+              }
+            ),
+            this.intl.t(
+              'decision-report-generation--toast-generating-complete--title'
+            ),
+            {
+              closable: true,
+              timeOut: 10 * 60 * 1000,
+            }
+          );
+      } else if (jobResult.status === CONSTANTS.DECISION_REPORT_JOB_STATUSSES.FAILURE) {
         this.toaster.error(
-          this.intl.t('error-while-generating-report-name-pdf', {
-            name: failure.report.name,
-            error: failure.error.message,
-          })
+          this.intl.t('error-while-generating-report-pdfs')
         );
+      } else {
+        setTimeout(() => {
+          this.pollReplacementReports.perform(job, reports, generatingPDFsToast);
+        }, 2000);
```

Comment:
**suggestion**: We have a `JobMonitor` service for stuff like this

It's not documented properly (you don't know how to use it without looking at where it's been previously used) but it's here: https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/development/app/services/job-monitor.js

An example of usage: https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/development/app/components/agenda/agenda-header/agenda-actions.js#L274-L322

Basically, you get a job record, register it to be monitored and pass a callback that will be called once the job is finished (which in practice is just used to replace the toaster).

Actually using it would require some substantial changes in the backend service so I think we can leave this as-is for now, but just something to keep in mind for the future.